Create new charter items for TD

egekorkan opened the pull request:
This is just a draft proposal (but not a draft PR). These are my collection of items that I have taken out of the issues of the TD spec and https://github.com/w3c/wot/issues/978. By choice, I do not go into details but explain enough so that the TF participants can remember the discussions. I think we have some work items that are not a single issue number. Once we agree on these generic work items, we can go into detail.

mlagally commented:
A couple of other suggestions:

* mechanisms to describe audio / video streams
* canonical form of a TD
* enhancements for the action and event model

mlagally commented:
TD 2.0 work should be guided by the use cases and requirements in https://www.w3.org/TR/wot-usecases.

benfrancis commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,34 @@
+
+Defining how TDs should be versioned, i.e. what does each version number incrementation mean for consumers.
+
+## Normative TD Parsing and Consuming
+
+See https://github.com/w3c/wot/issues/978#issuecomment-965127375
```
+1 I think TD 2.0 could benefit from a normative algorithm for parsing Thing Descriptions, in order to resolve various ambiguities.

mmccool replied:
For example, prefixes.  Also, standardizing validation, also for extensions.

egekorkan replied:
added validation to the title and detailed the text

benfrancis reviewed (approved):
One other thing which may not be big enough to mention here is a short form of security metadata. See https://github.com/w3c/wot-thing-description/issues/757

mmccool commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,34 @@
+
+How do we invoke, update, query, cancel actions and describe this in a generic way.
+
+## Payload Driven Protocols
+
+Some protocols are built on top of the application layer protocols. 
+Some are RPC protocols on top of HTTP, some use MQTT topics together with payloads to identify affordances.
```
Another one is JSON-RPC.  If not "standard", we may need to have a way to parameterize to actual use.  JSON-RPC may also need data models for messages.   Also Cloud sub/pub standards.

sebastiankb commented:
- guideline how to sign TDs (should based on existing standards)

mmccool commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,34 @@
+## Timeseries
+
+Some edge and cloud services can collect property value changes or event emissions for a certain time.
+This data, called timeseries, can be collected by Consumers to display (or use in any other way) the evolution of the affordance data over time.
```
May be related to discovery, e.g. may want to discover sensor values meeting certain criteria.  Also, devices may want to trigger events when certain things happen (or are in a certain state for a certain amount of time).

mmccool replied:
@egekorkan proposal is to add a query option to properties to get past values?

mmccool commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,34 @@
+
+## Externalizing Security Definitions
+
+See security proposal
```
Also simplifying, e.g. inline securitye defs

egekorkan replied:
added this as a separate field

mmccool commented:
Discussion:
dape: resolve profile and subset/superset of TD spec, pref for "subset", better interop
ml: TD descriptive, may need additional definitions; async action model seems to be main pain point; any others?
dape: should ANY profile superset TD?
ml: profile may specify additional constraints not in in TD spec, eg. behavior; also location
mccool: TDs do allow extension, so that is how I suggest we handle things like location; behaviour constraints are harder
bf: have used profiles to define things that are not describable in TDs; e.g. websocket protocols; even if make profiles subsets, we might need some other way to deal with defining such things (e.g. subprotocol)
kaz: really relationship between TD and profiles and other specs (e.g. protocol bindings).
ege: superset/subset not quite right phrasing; really about assumptions; should write protocol spec first
mccool: agree with kaz/ege, we should put the right specs in the right documents
seb: should position binding document so information is consolidated for recommended protocol usage; see for example DID structure for methods
bf: good to distinguish protocol binding template and a protocol binding; first is how to describe, second IS a description of a specific usage.  A profile is the second: a concrete spec; maybe we need a registry

mmccool commented:
General thought: alignment with DCAT standard (see https://www.w3.org/TR/vocab-dcat-2/) which was brought up by an attendee in the WoT CG meeting 13-09-2022.

This may or may not be the same as alignment with other RDF vocab standards (QUDT, SSN, ECLASS, iotschema, etc).  DCAT though is explicitly about metadata and data indexes, and seems relevant, and may be important in the Smart City context (we need to investigate further, however).

egekorkan commented:
@mlagally could you elaborate what you mean with "mechanisms to describe audio / video streams"? We are able to put a video or audio instance or its stream into a form element by using appropriate href and vocabulary terms . Do you mean that we should have binding templates for possible audio and video streams?

egekorkan commented:
@sebastiankb @mmccool @benfrancis @mlagally and others during the tpac call, thank you for your feedbacks! I have incorporated all of them except the point in the previous comment. I have also added related issues and PRs.

Please have another look :)

mmccool commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,97 @@
+Some examples are:
+- RPC protocols on top of HTTP such as JSON-RPC.
+- Protocols using MQTT topics together with payloads to identify affordances
+- Websocket sub-protocols
```
maybe cite web thing protocol cg document

mmccool commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,97 @@
+
+Related Issues:
+- https://github.com/w3c/wot-thing-description/issues/1023
+- https://github.com/w3c/wot-thing-description/issues/940
```
Topics to add:
- Streaming (distinct from time series; real-time, not historical)
- Consumer Descriptions (*maybe* separate from TDs; maybe a variant of an affordance; also related to MUDs)

mmccool replied:
Thing Models in its own document?  Improvements to it (e.g. modules?)

egekorkan commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,97 @@
+## Thing Description 2.0
+
```
I have added this, could you have a look?

egekorkan commented on proposals/deliverable-proposals/thing-description.md:
```diff
@@ -0,0 +1,107 @@
+- Webhook usage across implementations
+
+We should also decide and document in which cases this method does not work or make the TDs too complicated to understand.
+We cannot (and probably should not try to) describe the CoAP using a `udp://` uri scheme.
```
It's more for cases where the uri scheme is not really what the protocol is about. For example, XMLRPC uses http but has "stuff" around it. If we just put http uri scheme in the TD of a Thing that's actually using XMLRPC, the td would get complex. Same thing if you (for some weird reason) want to not use the coap uri scheme but put udp instead, you would need a probably more complex TD.
